### DISCUSSION TITLE
app.css failed to load. Please refresh the page, re-publish Log Viewer assets, or fix routing for vendor assets.

### DISCUSSION THREAD

**Discussion by syedali2415** [opcodesio/log-viewer#293] [General]
when i run php artisan serve i can open my log-viewer in http://127.0.0.1:8000/log-viewer
but not opening in https://dev.nikah.com/1rupee/1laraccali/log-viewer can anyone help me

**Comment by arukompas** [comment]
Hey @syedali2415 , is the **root** of your app at `https://dev.nikah.com/1rupee/1laraccali` ? Because I can see it's trying to load the vendor assets like it was root (e.g. `https://dev.nikah.com/1rupee/1laraccali/vendor/log-viewer/app.css`)

To me this seems like a server configuration issue. You need to check how your files are hosted, and how they're being served with apache/nginx. Looks like you have your app served on a sub-directory, and there might be a few things you need to do before your app works this way. I've never done this before, so I don't really have experience with that.

What have you found or tried so far?

**Reply by syedali2415** [reply]
yes as you said this might be the server configuration issue, it is working awesome i can see all my error logs individually with php artisan serve but when i run from https://dev.nikah.com/1rupee/1laraccali (my APP_URL) i cant able to get the logs first let me check with server side or anyother way to config this in log-viewer config please guide me and  very big thanks you saved me povilas(laravel_daily) suggest this package it is great this is what i am looking for... <3

**Reply by arukompas** [reply]
Hey @syedali2415 ,

Try hosting your app on the root level instead of a subfolder. That might be the issue. Also without seeing your server setup I can't really help, because there's thousands of ways to configure the server.

**Comment by nick-nds** [comment]
@syedali2415 , It seems like the issue you're facing with the log-viewer package arises from the configuration of the `ASSET_URL` in your `.env` file. When accessing the specified URL, you encounter an alert with an error regarding routing for vendor assets. Upon inspecting the network logs, it becomes evident that API calls made for `app.css` are returning a 404 error.

The root cause appears to be that the log-viewer package publishes its assets by default at the location `/public/vendor/log-viewer`. However, if you have set `ASSET_URL` to a different location, such as `<app-url>/assets`, then the `asset()` helper function used in the blade component will return `<app-url>/assets/vendor/log-viewer/app.js`. Consequently, the published `app.js` file exists at a different location, `<app-url>/vendor/log-viewer/app.js`.

To resolve this issue, there are two potential solutions:
  - Manually copy the log-viewer published assets to the location specified by your ASSET_URL.
  - Implement a patch in the package so that it considers the `ASSET_URL` while publishing assets.
  - Or, add proper url for log-viewer assets  in blade file.

@arukompas, if you believe that implementing this patch could resolve the issue, I would be willing to collaborate on it.

**Reply by arukompas** [reply]
@nick-nds the problem with `ASSET_URL` is usually used with CDNs and it is only meant to be used in front-facing urls, usually those created with `asset()` helper method. It does not tell the full story of where exactly **on disk** the files exist. It could be on a totally different machine, so it's dangerous to make any assumptions as it's likely going to be wrong.

**Comment by darkvovich** [comment]
I fixed it edited file /example-app/vendor/opcodesio/log-viewer/resources/views/index.blade.php
and reaplec all asset(mix( with secure_asset(mix(

But how to fix it NOT inside vendor packaige?

**Reply by arukompas** [reply]
@darkvovich hey, you can publish the layout (the file you just edited) by running `php artisan vendor:publish --tag=log-viewer-views`

See here - https://log-viewer.opcodes.io/docs/3.x/configuration/access-to-log-viewer#configuring-how-the-front-end-makes-api-requests

**Comment by denmely** [comment]
Add `.env`:

APP_URL=https://example.com
ASSET_URL=https://example.com
